icu4j/main/classes/core/src/com/ibm/icu/number/NumberFormatterImpl.java

```diff
@@ -353,14 +342,44 @@ private static MicroPropsGenerator macrosToMicroGenerator(MacroProps macros, Mic
                 // Lazily create PluralRules
                 rules = PluralRules.forLocale(macros.loc);
             }
-            chain = LongNameHandler
-                    .forMeasureUnit(macros.loc, macros.unit, macros.perUnit, unitWidth, rules, chain);
+            PluralRules pluralRules = macros.rules != null ?
+                    macros.rules :
+                    PluralRules.forLocale(macros.loc);
+
+            if (macros.usage != null) {
+                assert usagePrefsHandler != null;
+                chain = LongNameMultiplexer.forMeasureUnits(
+                        macros.loc,
+                        usagePrefsHandler.getOutputUnits(),
+                        unitWidth,
+                        pluralRules,
+                        chain);
```

Comment:
FYI: here the multiplexer is set up to point at the parent

Reply:
Ack.